### DISCUSSION TITLE
Team Leads and Managers Meeting 1

### DISCUSSION THREAD

**Discussion by david-baez-bravo** [uprm-inso4101-2024-2025-s1/semester-project-mercado-colegial-online#103] [General]
This thread is about what was discussed between managers and team leads in an online meeting on the 2nd of October 2024.

Things hopefully discussed:
- Team leads hold regular group meetings.
- Create GitHub Issues ASAP.
- Team Leads document managerial tasks.
- Discuss GitHub Issue for team leads.
- Check if team leads have already met with groups.
- Use GitHub discussions more.
- Ensure database is ready for login/signup.

Task-Based Goals del milestone:
- Crear e implementar un form para vendedores y consumidores del Mercado Colegial (Ayudara documentación y generación de data)
- Crear un dataset artificial para la base de datos (Pueden usar la herramienta MOSTLY AI o cualquier otro generador de data sintética)
- Conectar la base de datos al signup y login 

- Completar la funcionalidad del Signup/Login
- Crear un backend interface para el homepage que se encargue de enviar toda la data que la pagina necesite. (Un ejemplo claro sería los listings que aparecen el parte inferior)

- UI code implementation para el product listing page.
- Crear un backend interface para el product listing page que se encargue de enviar toda la data que la pagina necesite (Los listings de las categorías)

- UI code implementation para el product search page.
- Crear un backend interface para el product search page que se encargue de enviar toda la data que la pagina necesite (Los listings que sigan el search query)

**Comment by david-baez-bravo** [comment]
Documentation counts as creating a github discussion about meetings. Can be a link to pdf or google doc.

**Comment by david-baez-bravo** [comment]
The database is a ready for implementation with backend, need extensions for vs code. Needs to be populated.

Tasks for DB:

**Reply by david-baez-bravo** [reply]
- Implement the storing of reviews in the database

**Reply by YampierFernandez** [reply]
The database team’s tasks will focus on connecting to the data to the backend, making sure the database can be populated, accessed and edited from the application instead of only through Atlas. The application should be able to perform functions such as get, pull, post, delete, etc. The necessary extensions to make this possible must applied and documented. The models will also be updated to better mix with the current needs of the application.

Many tasks will have to be done together with the backend team. Task that are better suited for both backend and database team members must be detailed as such so managers can know to assign members from both teams.

**Comment by david-baez-bravo** [comment]
Tools for database integration with backend:

**Reply by ActuallySeb** [reply]
Ajax is an existing functionality of JS that allows the application to access/update the server's data without refreshing the page.
https://www.w3schools.com/xml/ajax_intro.asp

Fetch is an API that does the same thing based on a upgraded version of how AJAX works. It's a little harder to maneuver around the fact that it reloads the page with every request made, but there are ways to prevent it.
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

**Comment by YampierFernandez** [comment]
Model for items:


```
const mongoose = require('mongoose');

const itemSchema = new mongoose.Schema({

  item_id: {
    type: mongoose.Schema.Types.ObjectId, 
    index: true,
    unique: true,
  },

  item_name: {
    type: String,
    required: true,
    maxlength: 255,  
  },
 
  price: {
    type: Number,
    required: true,
    min: 0,  // Price cannot be negative
  },
 
  stock: {
    type: Number,
    required: true,
    min: 0,  // Stock cannot be negative
  },
 
  description: {
    type: String,
    maxlength: 1000,  
  },
}, {
  timestamps: true  
});

// Create and export the model based on the schema
const Item = mongoose.model('Item', itemSchema);

module.exports = Item;
```

Model for users:

```
const mongoose = require('mongoose');

const userSchema = new mongoose.Schema({
  // Mongoose automatically creates an `_id` field (ObjectId), so no need to add `user_id`
  
  username: {
    type: String,
    required: true,
    trim: true,
    minlength: 3,
    maxlength: 50,
  },
 
  email: {
    type: String,
    required: true,
    unique: true,
    trim: true,
    lowercase: true,
    match: [/\S+@\S+\.\S+/, 'is invalid'], // Simple email validation
  },
 
  password: {
    type: String,
    required: true,
    minlength: 8, // Enforcing password length
  },

  student_num: {
    type: Number,
    required: true,
    unique: true, // Assuming each student number is unique
  },
 
  role: {
    type: String,
    required: true,
    enum: ['student', 'admin', 'guest'], // Optional role validation
  },

  card_num: {
    type: String, // Changed to string to avoid losing leading zeros
    required: true,
    minlength: 13,
    maxlength: 16,
  },

  card_name: {
    type: String,
    required: true,
    trim: true,
  },

  cvv: {
    type: String, // Changed to string for leading zeros and consistent length
    required: true,
    minlength: 3,
    maxlength: 3,
  },

  card_zip: {
    type: String, // Changed to string for leading zeros
    required: true,
    minlength: 4,
    maxlength: 5, // Zip codes can vary (US zip codes have 5 digits)
  },

  shipping_zip: {
    type: String, // Changed to string for leading zeros
    required: true,
    minlength: 4,
    maxlength: 5,
  },

  addr_line_1: {
    type: String,
    required: true,
    trim: true,
  },

  addr_line_2: {
    type: String,
    trim: true,
    default: '', // Optional field
  },

  city: {
    type: String,
    required: true,
    trim: true,
  },

}, {
  timestamps: true  // Automatically create `createdAt` and `updatedAt` fields
});

const User = mongoose.model('User', userSchema);

module.exports = User;
```

**Comment by david-baez-bravo** [comment]
Tasks Web dev:
- Fix the color scheme of web pages
